src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java

```diff
@@ -1369,16 +1287,146 @@ else if (isPostGISDataStore(dataStore)) {
                 if (error != null) {
                     error = e;
                 }
-                LOGGER.log(Level.WARNING, "Error closing transaction",e);
+                LOGGER.log(Level.WARNING, "Error closing transaction", e);
             }
         }
-        
         // finally, throw any error
         if (error != null) {
             throw error;
         }
     }
 
+    private Exception copyFeaturesHighLevelAPI(ImportData data, ImportTask task,
+            DataStoreFormat format, DataStore dataStore, Transaction transaction,
+            String featureTypeName, String uniquifiedFeatureTypeName,
+            FeatureDataConverter featureDataConverter, VectorTransformChain tx) throws IOException {
```

Comment:
Would prefer "copyFromFeatureSource" since we do not have a formal high / low / internal api.

Is it worth untangling some of these featureTypeName, dataStore, transaction parameters by passing in a single configured featureStore to this method?